### PR TITLE
feat: flux cd resources

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
feat: add support fro missing features in flux cd app details
- Delete Resource
- Vulnerabilities Status
- Connect Terminal, Edit Manifest, Get Logs, Download Logs, Get Url, Scale Workloads

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/appDetails.api.ts [hunk 1]
```diff
@@ -48,28 +49,28 @@ export const deleteResource = (nodeDetails: any, appDetails: any, envId: string,
...
-    const { group, version, kind, name } = nodeDetails
+    const { appName, deploymentAppType, clusterId, namespace, appType, appId, fluxTemplateType } = appDetails
+    const { group, version, kind, name, namespace: nodeNamespace } = nodeDetails
 
     const data = {
         appId:
             appType == AppType.DEVTRON_APP
                 ? generateDevtronAppIdentiferForK8sRequest(clusterId, appId, Number(envId))
-                : getAppId(clusterId, namespace, appName),
+                : getAppId(clusterId, namespace, appName, fluxTemplateType ?? null),
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/appDetails.api.ts:59]: Can move params to object.

File: src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts [hunk 2]
```diff
@@ -17,23 +17,25 @@
...
-} from '../../appDetails.type'
+import { AppDetails, AppType, SelectedResourceType } from '../../appDetails.type'
 import { ParamsType } from './nodeDetail.type'
-import { getK8sResourcePayloadAppType } from './nodeDetail.util'
+import { getDeploymentType, getK8sResourcePayloadAppType } from './nodeDetail.util'
 
-export const getAppId = (clusterId: number, namespace: string, appName: string) => {
+export const getAppId = (clusterId: number, namespace: string, appName: string, templateType?: string) => {
+    if (templateType) {
+        return `${clusterId}|${namespace}|${appName}|${templateType === 'Kustomization'}`
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts:26]: Use enum

File: src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.util.ts [hunk 3]
```diff
@@ -485,6 +486,9 @@ export const getK8sResourcePayloadAppType = (appType: string) => {
     if (appType === AppType.EXTERNAL_ARGO_APP) {
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.util.ts:486]: Can add return type of K8sResourcePayloadAppType to function

File: src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.util.ts [hunk 4]
```diff
@@ -498,3 +502,13 @@ export const getDecodedEncodedSecretManifestData = (
     }
     return returnAsString ? JSON.stringify(encodedData) : manifestData
 }
+
+export const getDeploymentType = (deploymentAppType: DeploymentAppTypes) => {
+    if (deploymentAppType === DeploymentAppTypes.HELM) {
+        return K8sResourcePayloadDeploymentType.HELM_INSTALLED
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.util.ts:508]: Can add return type to method of K8sResourcePayloadDeploymentType

File: src/components/v2/appDetails/AppDetails.component.tsx [hunk 5]
```diff
@@ -94,8 +94,8 @@ const AppDetailsComponent = ({
     useEffect(() => {
         // Get deployment status timeline on argocd apps
         if (
-            appDetails?.deploymentAppType === DeploymentAppTypes.GITOPS ||
-            appDetails?.deploymentAppType === DeploymentAppTypes.MANIFEST_DOWNLOAD
+            (appDetails?.deploymentAppType === DeploymentAppTypes.GITOPS ||
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/AppDetails.component.tsx:97]: can create a variable with meaningful name like isHelmApp

File: src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx [hunk 6]
```diff
@@ -350,7 +359,9 @@ const EnvironmentSelectorComponent = ({
                                 deployedAppDetail[2],
                                 deployedAppDetail[1],
                                 deployedAppDetail[0],
-                            ))
+                            )) ||
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx:362]: Please add comment for this condition

File: src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.util.ts [hunk 7]
```diff
@@ -26,13 +26,14 @@ import {
...
     NodeType,
+    K8sResourcePayloadDeploymentType,
 } from '../../appDetails.type'
 import IndexStore from '../../index.store'
 import { ManifestData, NodeDetailTab } from './nodeDetail.type'
 import { multiSelectStyles } from '../../../common/ReactSelectCustomization'
 import { sortOptionsByLabel } from '../../../../common'
 import { ALLOW_UNTIL_TIME_OPTIONS, CUSTOM_LOGS_FILTER, MANIFEST_KEY_FIELDS } from '../../../../../config'
-import { decode } from '@devtron-labs/devtron-fe-common-lib'
+import { decode, DeploymentAppTypes } from '@devtron-labs/devtron-fe-common-lib'
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.util.ts:36]: can move to top

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts [hunk 8]
```diff
@@ -410,14 +367,15 @@ export const deleteEphemeralUrl = (
     appName: string,
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts:367]: Can we transform params into object for deleteEphemeralUrl and generateEphemeralUrl

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**